contracts/airdrop.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.5.15;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {IERC20} from "./interfaces/IERC20.sol";
+import {ITokenRegistry} from "./interfaces/ITokenRegistry.sol";
+
+import {Types} from "./libs/Types.sol";
+import {RollupUtils} from "./libs/RollupUtils.sol";
+import {ECVerify} from "./libs/ECVerify.sol";
+
+import {MerkleTreeUtils as MTUtils} from "./MerkleTreeUtils.sol";
+
+contract AirdropSetup {
+    using SafeMath for uint256;
+    using ECVerify for bytes32;
+
+    MTUtils public merkleUtils;
+
+    bytes32
+        public constant ZERO_BYTES32 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+
+    /********************
+     * Error Codes *
+     ********************/
+    uint256 public constant NO_ERR = 0;
+    uint256 public constant ERR_TOKEN_ADDR_INVAILD = 1; // account doesnt hold token type in the tx
+    uint256 public constant ERR_TOKEN_AMT_INVAILD = 2; // tx amount is less than zero
+    uint256 public constant ERR_TOKEN_NOT_ENOUGH_BAL = 3; // leaf doesnt has enough balance
+    uint256 public constant ERR_FROM_TOKEN_TYPE = 4; // from account doesnt hold the token type in the tx
+    uint256 public constant ERR_TO_TOKEN_TYPE = 5; // to account doesnt hold the token type in the tx
+}
+
+contract AirdropHelpers is AirdropSetup {
+    function AddTokensToAccount(
+        Types.UserAccount memory account,
+        uint256 numOfTokens
+    ) public pure returns (Types.UserAccount memory updatedAccount) {
+        return (
+            RollupUtils.UpdateBalanceInAccount(
+                account,
+                RollupUtils.BalanceFromAccount(account).add(numOfTokens)
+            )
+        );
+    }
+    function UpdateAccountWithSiblings(
+        Types.UserAccount memory new_account,
+        Types.AccountMerkleProof memory _merkle_proof
+    ) public view returns (bytes32, uint256) {
+        bytes32 newRoot = merkleUtils.updateLeafWithSiblings(
+            keccak256(RollupUtils.BytesFromAccount(new_account)),
+            _merkle_proof.accountIP.pathToAccount,
+            _merkle_proof.siblings
+        );
+        uint256 balance = RollupUtils.BalanceFromAccount(new_account);
+        return (newRoot, balance);
+    }
+}
+
+contract Airdrop is AirdropHelpers {
+
+    function processDrop(
+        Types.Transaction memory drop,
```

Comment:
We parse the same transaction format but we care only the `to` and `amount` field only

Reply:
One optimization for later is to have a differnet transatino format so that we don't need to put from on chain.